packages/connection-storage/src/connection-storage.ts

```diff
@@ -1,520 +1,170 @@
-    // If user lands on this version of Compass with legacy connections (using storage mixin),
-    // we will ignore those and only migrate connections that have connectionInfo.
-    const connections = (await this.getConnections()).filter(
-      (
-        x
-      ): x is ConnectionWithLegacyProps &
-        Required<Pick<ConnectionProps, 'connectionInfo'>> => !!x.connectionInfo
-    );
-    if (connections.length === 0) {
-      log.info(
-        mongoLogId(1_001_000_270),
-        'Connection Storage',
-        'No connections to migrate'
-      );
-      return;
-    }
+  off<T extends ConnectionStorageEvent>(
+    eventName: T,
+    listener: ConnectionStorageEventListeners[T]
+  ): ConnectionStorage;
 
-    if (
-      connections.filter((x) => 'version' in x && x.version === this.version)
-        .length === connections.length
-    ) {
-      log.info(
-        mongoLogId(1_001_000_271),
-        'Connection Storage',
-        'Connections already migrated'
-      );
-      return;
-    }
+  emit<T extends ConnectionStorageEvent>(
+    eventName: T,
+    ...args: Parameters<ConnectionStorageEventListeners[T]>
+  ): boolean;
 
-    // Get the secrets from keychain. If there're no secrets, we still want to
-    // migrate the connects to a new version.
-    const secrets = await this.getKeytarCredentials();
+  loadAll(options?: { signal?: AbortSignal }): Promise<ConnectionInfo[]>;
 
-    // Connections that are not migrated yet or that failed to migrate,
-    // are not expected to have a version property.
-    const unmigratedConnections = connections.filter((x) => !('version' in x));
-    log.info(
-      mongoLogId(1_001_000_272),
-      'Connection Storage',
-      'Migrating connections',
-      {
-        num_connections: unmigratedConnections.length,
-        num_secrets: Object.keys(secrets).length,
-      }
-    );
+  load(options: {
+    id: string;
+    signal?: AbortSignal;
+  }): Promise<ConnectionInfo | undefined>;
 
-    let numFailedToMigrate = 0;
+  save?(options: {
+    connectionInfo: ConnectionInfo;
+    signal?: AbortSignal;
+  }): Promise<void>;
 
-    for (const { connectionInfo } of unmigratedConnections) {
-      const _id = connectionInfo.id;
-      const connectionSecrets = secrets[_id];
+  delete?(options: { id: string; signal?: AbortSignal }): Promise<void>;
+}
 
-      try {
-        await this.userData.write(_id, {
-          _id,
-          connectionInfo, // connectionInfo is without secrets as its direclty read from storage
-          connectionSecrets: this.encryptSecrets(connectionSecrets),
-          version: this.version,
-        });
-      } catch (e) {
-        numFailedToMigrate++;
-        log.error(
-          mongoLogId(1_001_000_273),
-          'Connection Storage',
-          'Failed to migrate connection',
-          { message: (e as Error).message, connectionId: _id }
-        );
-      }
-    }
+export interface CompassConnectionStorage extends ConnectionStorage {
```

Comment:
I'd probably suggest not to split this in two and instead make them optional on the single interface and just account for this in the client code (similar to save and delete)